### PR TITLE
ASimNetworkSessionMicrosoftSysmon.yaml-42

### PR DESCRIPTION
**Pull request by Alekhya0824**
Required items, please complete
   
   Change(s):
   - Created a new file for WindowsEvent and added WindowsEvent parser in it. In Wvent file removed WindowsEvent parser and in ASimNetworkSession added WindowsEvent Parser name .

   Reason for Change(s):
   - To Split the WindowsEvent and Event parser separately

   Version Updated:
   - yes

   Testing Completed:
   - yes

   Checked that the validations are passing and have addressed any issues that are present:
   - yes

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
KQL validation failures, please check for those. Thanks
image

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Validations failing please check. Thanks

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please clear failed validations. Thanks

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: Network Session Event ASIM parser for Sysmon (Event 3)
-  Version: '0.1'
-  LastUpdated: Jan 11, 2023
+  Version: '0.2.1'
+  LastUpdated: June 5, 2024
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:4]: please correct date

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ References:
 - Title: ASIM
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:13]: ASIM schema "NetworkSession" version "0.2.5" is incorrect. The correct version for ASIM schema "NetworkSession" is "0.2.6"

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ References:
...
   Link: https:/aka.ms/AboutASIM
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:14]: url correction

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  Version: '0.2.5'
+References:
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+Description: |
+  This ASIM parser supports normalizing Sysmon network session events (event 3) collected using the Event or WEF (WindowsEvent table) connectors logs ingested in 'WindowsEvent' table to the ASIM Process Event normalized schema. 
+ParserName: vimNetworkSessionMicrosoftSysmonWindowsEvent
+EquivalentBuiltInParser: _im_NetworkSession_MicrosoftSysmonWindowsEvent
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:18]: _Im_

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 5]
```diff
@@ -0,0 +1,184 @@
+Parser:
+  Title: Network Session Event ASIM parser for Sysmon (Event 3)
+  Version: '0.1.1'
+  LastUpdated: June 20, 2024
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:4]: date correction

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+Product:
+  Name: Windows Sysmon
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.5'
+References:
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:14]: url correction

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 7]
```diff
@@ -0,0 +1,139 @@
+Parser:
+  Title: Network Session Event ASIM parser for Sysmon (Event 3)
+  Version: '0.2.1'
+  LastUpdated: June 5, 2024
+Product:
+  Name: Windows Sysmon
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.5'
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:9]: ASIM schema "NetworkSession" version "0.2.5" is incorrect. The correct version for ASIM schema "NetworkSession" is "0.2.6"

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 8]
```diff
@@ -0,0 +1,139 @@
+Parser:
+  Title: Network Session Event ASIM parser for Sysmon (Event 3)
+  Version: '0.2.1'
+  LastUpdated: June 5, 2024
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:4]: date correction

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+Product:
+  Name: Windows Sysmon
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.5'
+References:
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:14]: url correction

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+  Version: '0.2.5'
+References:
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+Description: |
+  This ASIM parser supports normalizing Sysmon network session events (event 3) collected using the Event or WEF (WindowsEvent table) connectors logs ingested in 'Event' table to the ASIM Process Event normalized schema. 
+ParserName: vimNetworkSessionMicrosoftSysmon
+EquivalentBuiltInParser: _im_NetworkSession_MicrosoftSysmon
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml:18]: _Im_

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml [hunk 11]
```diff
@@ -0,0 +1,192 @@
+Parser:
+  Title: Network Session Event ASIM parser for Sysmon (Event 3)
+  Version: '0.1.1'
+  LastUpdated: June 18, 2024
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml:4]: date correction

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+Product:
+  Name: Windows Sysmon
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.5'
+References:
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml:14]: url correction

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 13]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:1]: (1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 14]
```diff
@@ -0,0 +1,139 @@
+Parser:
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:1]: (1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml [hunk 15]
```diff
@@ -0,0 +1,192 @@
+Parser:
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml:1]: (1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 16]
```diff
@@ -0,0 +1,184 @@
+Parser:
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:1]: (1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml [hunk 17]
```diff
@@ -0,0 +1,192 @@
...
+        ipaddr_has_any_prefix:dynamic=dynamic([]),
+        dstportnumber:int=int(null), 
+        hostname_has_any:dynamic=dynamic([]), 
+        dvcaction:dynamic=dynamic([]), 
+        eventresult:string='*', 
+        disabled:bool=false
+        )
+        {
+        let Sysmon3_Event=Event
+        | where (isnull(starttime) or TimeGenerated >= starttime) and  (isnull(endtime) or TimeGenerated <= endtime)
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml:65]: we need to use all ASIM networkSession filters in this parser

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 18]
```diff
@@ -0,0 +1,184 @@
...
+        {
+        let Sysmon3_WindowsEvent=WindowsEvent
+        | where not(disabled) 
+        | where (isnull(starttime) or TimeGenerated >= starttime) and  (isnull(endtime) or TimeGenerated <= endtime)
+        // -- Pre-filtering:
+        | where
+            eventresult  in ("*", "Success") 
+            and array_length(dvcaction) == 0
+            and  dstportnumber==int(null) 
+            and (array_length(hostname_has_any)==0 or Computer has_any (hostname_has_any))
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:72]: we need to use all ASIM networkSession filters in this parser

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSession.yaml [hunk 19]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: Network Session ASIM parser
-  Version: '0.7'
-  LastUpdated: Jan 17, 2024
+  Version: '0.7.2'
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSession.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 20]
```diff
@@ -1,7 +1,7 @@
...
   Title: Network Session Event ASIM parser for Sysmon (Event 3)
-  Version: '0.1'
-  LastUpdated: Jan 11, 2023
+  Version: '0.2.1'
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 21]
```diff
@@ -13,7 +13,7 @@ References:
 - Title: ASIM
   Link: https:/aka.ms/AboutASIM
 Description: |
-  This ASIM parser supports normalizing Sysmon network session events (event 3) collected using the Event or WEF (WindowsEvent table) connectors to the ASIM Process Event normalized schema. 
+  This ASIM parser supports normalizing Sysmon network session events (event 3) collected using the Event or WEF (WindowsEvent table) connectors logs ingested in 'Event' table to the ASIM Process Event normalized schema.
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 22]
```diff
@@ -0,0 +1,139 @@
...
+  This ASIM parser supports normalizing Sysmon network session events (event 3) collected using the Event or WEF (WindowsEvent table) connectors logs ingested in 'WindowsEvent' table to the ASIM Process Event normalized schema. 
+ParserName: ASimNetworkSessionMicrosoftSysmonWindowsEvent
+EquivalentBuiltInParser: _ASim_NetworkSession_MicrosoftSysmonWindowsEvent
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: | 
+  let parser = (disabled:bool = false) {
+  let Sysmon3_WindowsEvent=(disabled:bool=false){
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:25]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 23]
```diff
@@ -13,7 +13,7 @@ References:
 - Title: ASIM
   Link: https:/aka.ms/AboutASIM
 Description: |
-  This ASIM parser supports normalizing Sysmon network session events (event 3) collected using the Event or WEF (WindowsEvent table) connectors to the ASIM Process Event normalized schema. 
+  This ASIM parser supports normalizing Sysmon network session events (event 3) collected using the Event or WEF (WindowsEvent table) connectors logs ingested in 'Event' table to the ASIM Process Event normalized schema. 
 ParserName: ASimNetworkSessionMicrosoftSysmon
 EquivalentBuiltInParser: _ASim_NetworkSession_MicrosoftSysmon
 ParserParams:
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:19]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml (hunk 1 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml (hunk 2 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:13]: Done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 4 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 5 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 6 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 7 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:9]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 8 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 9 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml (hunk 10 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml (hunk 11 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml (hunk 12 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml (hunk 13 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:1]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 14 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:1]: Done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 16 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:1]: Done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSession.yaml [hunk 24]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSession.yaml:1]: I Didn't got

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
I can see a few open comments from the reviewer. Please act on them.
image
And also, one suggestion: once you're done with the open comment change, please click on the `Resolve Conversation` button.

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSession.yaml (hunk 24 above)
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSession.yaml:1]: updating table schema file

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 25]
```diff
@@ -62,39 +62,7 @@ ParserQuery: |
...
-        | project-away
-                       Provider,
-                       Channel,
-                       Task,
-                       Data,
-                       RawEventData,
-                       EventOriginId
-      };
-  union isfuzzy=true Sysmon3_Event,Sysmon3_WindowsEvent
+  Sysmon3_Event
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml]: done

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please check and respond on above comments. Thanks

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 26]
```diff
@@ -1,19 +1,19 @@
 Parser:
   Title: Network Session Event ASIM parser for Sysmon (Event 3)
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:2]: @Alekhya0824 is this a duplicate PR?
https://github.com/Azure/Azure-Sentinel/pull/10571

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml (hunk 17 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml]: done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 18 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:75]: done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml (hunk 17 above)
Comment (vakohl) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml]: Filtering is incorrect

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml [hunk 27]
```diff
@@ -0,0 +1,196 @@
...
+        let Sysmon3_Event=Event
+        | where not(disabled)
+        | where (isnull(starttime) or TimeGenerated >= starttime) and  (isnull(endtime) or TimeGenerated <= endtime)
+        | where (array_length(src_or_any)==0) and (array_length(dst_or_any)==0)
+        // -- Pre-filtering:
+        | where
+            eventresult  in ("*", "Success") 
+            and array_length(dvcaction) == 0
+            and  dstportnumber==int(null) 
+            and (array_length(hostname_has_any)==0 or Computer has_any (hostname_has_any))
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml:75]: filtering seems incorrect, did we test post changes?

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**